libr/util/unum.c

```diff
@@ -75,31 +75,37 @@ R_API void r_num_free(RNum *num) {
 	R_FREE (num);
 }
 
-#define KB (1024)
-#define MB (1024*KB)
-#define GB (1024*MB)
-#define TB (1024*GB)
-
+#define KB (((ut64)1) << 10)
+#define MB (((ut64)1) << 20)
+#define GB (((ut64)1) << 30)
+#define TB (((ut64)1) << 40)
+#define PB (((ut64)1) << 50)
+#define EB (((ut64)1) << 60)
+
+// TODO: Use SI system
+// https://metacpan.org/pod/Number::Bytes::Human
 R_API char *r_num_units(char *buf, ut64 num) {
 	char unit;
 	int tnum;
 	double fnum = num;
-	if (!buf) {
-		buf = malloc (32);
-		if (!buf) {
+	if (buf == NULL) {
```

Comment:
== NULL is. not necessary. just use !buf like it was before

Reply:
[FreeBSD prefer testing explicit for NULL](https://www.freebsd.org/cgi/man.cgi?query=style&sektion=9&manpath=freebsd-release-ports).

Quoted:

```
Test pointers against NULL, e.g., use:
     (p	= f()) == NULL
not:
     !(p = f())

Do not use ! for tests unless it is a boolean, e.g. use:
     if	(*p == '\0')
not:
     if	(!*p)
```

I think it's more readable. But I change it as you ask.

Reply:
As with any project, unfortunately, each one has its own set of rules and style. In r2 we do use `!` to check for NULL. You may find it less readable (I do prefer `== NULL` as well, actually) but the important thing is to be consistent with the rest of the code, which use `!`.